packages/website/docs/contributing/guides/1_adding-a-component.mdx

```diff
@@ -0,0 +1,457 @@
+---
+sidebar_label: New Components
+---
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+# Adding a new component to Unovis
+This guide will take you through the steps of adding a new component to our library.
+
+## Prerequisites
+Before you start, make sure you familiarize yourself with our library's
+[architecture](./architecture). You should also be comfortable working with
+[TypeScript](https://www.typescriptlang.org) and [d3](https://d3js.org).
+
+## Getting started
+### SVG vs HTML components
+SVG components are rendered as `g` elements, and therefore are designed to exist inside of a `svg`
+container (_XYContainer_ or _SingleContainer_). HTML elements are designed to stand alone. In
+general, "core" components that display data (i.e. charts, plots and diagrams) will be SVGs and
+auxiliary components (i.e. annotations, legends and tooltips) will be HTML. Note the following
+exceptions:
+
+* A core component will be type HTML in special cases, for example when HTML canvas is needed for
+rendering (i.e. LeafletMap)
+* An auxiliary component should be SVG if it is designed to exclusively work with XY data,
+in which case it should extend the _XYComponentCore_ superclass which uses SVG type
+
+### Data Types
+An important thing to consider beforehand is your component's expected data type. This will
+determine the component's superclass (if applicable), the corresponding `datamodel` and generics.
+
+You can view the currently supported data models in `packages/ts/data-models` directory. For the
+purpose of this guide, we will group data types in accordingly:
+
+* **XY Data ** - data should be an array of type `any[]`, designed for to be plotted on XY coordinate system
+* **Series Data ** - an array of arbitrary values; type `any[]`
```

Comment:
Maybe it's worth adding a few sentences that the data format is not strictly defined because we actively using accessor functions throughout the library.